File: src/lib/software/itunes.py
from src.lib.software import BaseSoftware

class ITunes(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'iTunes64Setup.exe'

    def resolve_download_url(self):
        self.url_resolved.emit('https://www.apple.com/itunes/download/win64')

File: packages/app/src/services/software/definitions/itunes.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'itunes';
	public name = 'iTunes';
	public category = [SoftwareCategory.Audio, SoftwareCategory.Media];
	public downloadName = 'iTunes64Setup.exe';
	public icon = 'itunes.png';
	public homepage = 'https://apple.com/itunes';

	public async resolveDownloadUrl() {
		return ok('https://www.apple.com/itunes/download/win64');
	}
}
